WP_DEBUG: REST API LocationsRoute und ProjectsRoute Schema Validation

CB Version: 2.8.5
WP Version 6.4
PHP 8.1

**Describe the bug**
Sowohl die LocationsRoute als auch die ProjectsRoute der REST API rufen die schemaValidation Methode im WP Debug Modus auf. Die zugehörenden Schema Dateien sind aber nicht Bestandteil des Plugins, sodass die Aufrufe mit Fehler beendet werden.

**To Reproduce**
WP_DEBUG einschalten und z.B. http....../wp-json/commonsbooking/v1/location aufrufen

**Expected behavior**
Ausgabe einer Json Datei

**Screenshots**

Beim ausführen von  bin/build-zip.sh bekomme ich:

Generating build directory...
Installing PHP and JS dependencies...
npm ERR! Cannot read property '@commonsbooking/frontend' of undefined

Auch nach Löschen des node_modules Ordners der Fehler (bei mir übrigens nicht Docker sonder WSL2 mit Ubuntu 20.04):

Loading "dart-sass.js" tasks...ERROR
>> ReferenceError: globalThis is not defined
Warning: Task "dart-sass:admin" not found. Use --force to continue.

Aborted due to warnings.
npm ERR! code ELIFECYCLE
npm ERR! errno 3
npm ERR! commonsbooking@2.8.5 dist: `grunt dist`
npm ERR! Exit status 3
npm ERR!
npm ERR! Failed at the commonsbooking@2.8.5 dist script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

--- a/src/API/BaseRoute.php
+++ b/src/API/BaseRoute.php
@@ -90,6 +90,12 @@ class BaseRoute extends WP_REST_Controller {
 	 */
 	public function validateData( $data ) {
 		$validator = new Validator();
+		//the validation routes are not shipped with the built plugin
+		$isDevEnvironment = file_exists( COMMONSBOOKING_PLUGIN_DIR . 'node_modules' );
+
+		if ( ! $isDevEnvironment ) {
+			return;
+		}
 
 		try {
 			$result = $validator->schemaValidation( $data, $this->getSchemaObject() );

--- a/src/API/ItemsRoute.php
+++ b/src/API/ItemsRoute.php
@@ -99,6 +99,10 @@ class ItemsRoute extends BaseRoute {
 
 		}
 
+		if ( WP_DEBUG ) {
+			$this->validateData( $data );
+		}
+
 		return new WP_REST_Response( $data, 200 );
 	}
 

